registry/proxy/proxyblobstore.go

```diff
@@ -0,0 +1,214 @@
+package proxy
+
+import (
+	"io"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/registry/proxy/scheduler"
+)
+
+// todo(richardscothern): from cache control header or config file
+const blobTTL = time.Duration(24 * 7 * time.Hour)
+
+type proxyBlobStore struct {
+	localStore  distribution.BlobStore
+	remoteStore distribution.BlobService
+	scheduler   *scheduler.TTLExpirationScheduler
+}
+
+var _ distribution.BlobStore = proxyBlobStore{}
```

Comment:
It asserts that proxyBlobStore implements the distribution.BlobStore
interface.

On Tue, Aug 4, 2015 at 3:46 PM, Aaron Lehmann <notifications@github.com>
wrote:

> In registry/proxy/proxyblobstore.go
> <https://github.com/docker/distribution/pull/779#discussion_r36251728>:
>
> > +	"github.com/docker/distribution"
> > +	"github.com/docker/distribution/context"
> > +	"github.com/docker/distribution/digest"
> > +	"github.com/docker/distribution/registry/proxy/scheduler"
> > +)
> > +
> > +// todo(richardscothern): from cache control header or config file
> > +const blobTTL = time.Duration(24 * 7 * time.Hour)
> > +
> > +type proxyBlobStore struct {
> > +	localStore  distribution.BlobStore
> > +	remoteStore distribution.BlobService
> > +	scheduler   *scheduler.TTLExpirationScheduler
> > +}
> > +
> > +var _ distribution.BlobStore = proxyBlobStore{}
>
> What does this line do?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/distribution/pull/779/files#r36251728>.
>